admin/config

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash -eu
+
+#
+# BMC config backup/restore.
+#
+
+# prevent modification of executable path
+PATH="/bin:/sbin:/usr/bin:/usr/sbin"
+
+# elevate self privileges
+[[ ${EUID} -eq 0 ]] || exec sudo $(realpath ${BASH_SOURCE[0]}) $@
+
+# list of files to save/restore
+WHITELIST="/var/run/initramfs/whitelist"
+
+# parse command line options
+ACTION="help"
+[[ $# -eq 0 ]] || ACTION="$1"
+case "${ACTION}" in
+  help | -h | --help )
+    echo "Backup or restore OpenBMC configuration."
+    echo "Use: ${0##*/} [OPTION...]"
+    echo "  backup FILE   Save configuration to file"
+    echo "  restore FILE  Restore configuration from file"
+    echo "  reset         Reset configuration to manufacturing default"
+    echo "  -h, --help    Print this help and exit"
+    ;;
+
+  backup )
+    [[ $# -eq 2 ]] || { echo "Invalid arguments" >&2; exit 1; }
+    tar czf "$2" -T ${WHITELIST} &>/dev/null || /bin/true
+    if [[ -f $2 ]]; then
+      echo "Backup created: $2"
+    else
+      echo "Error creating backup: $2" >&2
+      exit 1;
+    fi
+    ;;
+
+  restore )
+    [[ $# -eq 2 ]] || { echo "Invalid arguments" >&2; exit 1; }
+    TMP_DIR="$(mktemp -d)"
+    tar xf "$2" -C ${TMP_DIR}
+    for FILE in $(cat ${WHITELIST}); do
+      [[ -e ${TMP_DIR}${FILE} ]] || continue
+      rm -rf ${FILE} &>/dev/null || true
```

Comment:
As I [said](https://github.com/YADRO-KNS/obmc-yadro-cli/pull/1#discussion_r412824747), complex logic requires more flexible language.
About getopt - it's a headache for linux, we have getopt and getopts, but both are ugly, one doesn't support long options, another is bashism and requires `eval`, which is not safe in pair with sudo.

Reply:
Nothing stops us from using C or C++ here if that is needed. Server's serial number can well be checked in bash though  using `lsinventory` or `fru` from `phosphor-clio`.

Reply:
The main problem is in services, that uses the non-volatile storage. There are not only config files, but also caches uses in runtime. Some of services may change own files while we trying to save/restore them and it may lead to unpredictable behavior.